### ISSUE TITLE
Wireframe / design profile

### ISSUE THREAD

**Issue by rnystrom** [GitHawkApp/GitHawk#2106]

**Comment by Huddie** [comment]
I am not a graphic designer but maybe these can get the discussion sparked here.


![profile stage 1](https://user-images.githubusercontent.com/10662471/44243868-3708dd80-a19f-11e8-86ae-414339791c1d.jpg)

![profile stage 3](https://user-images.githubusercontent.com/10662471/44243869-3a03ce00-a19f-11e8-8a8c-8a1d22259774.jpg)

![profile stage 4](https://user-images.githubusercontent.com/10662471/44243873-3c662800-a19f-11e8-8a71-5c569df49437.jpg)

**Comment by Huddie** [comment]
I'll also point out. I don't really like that bookmarks are the center icon on the tabbar. For some reason I feel it should be profile or inbox. But I wasn't really focused on that

**Comment by BrianLitwin** [comment]
Including the following images here for quick reference against what Github already does. It would be cool to find some fresh sources of inspiration beyond GitHub. 

---------------------------------------------------------------------------------------------------
<img width="356" alt="screenshot 2018-08-15 16 02 06" src="https://user-images.githubusercontent.com/26695477/44244164-8f8caa80-a1a0-11e8-9e84-132b2655e35c.png">
---------------------------------------------------------------------------------------------------
<img width="355" alt="screenshot 2018-08-15 16 02 53" src="https://user-images.githubusercontent.com/26695477/44244169-99161280-a1a0-11e8-9ae0-396c5fa29f21.png">
---------------------------------------------------------------------------------------------------
<img width="353" alt="screenshot 2018-08-15 16 01 49" src="https://user-images.githubusercontent.com/26695477/44244215-cebafb80-a1a0-11e8-900f-014e5154c4a0.png">
---------------------------------------------------------------------------------------------------
<img width="398" alt="screenshot 2018-08-16 22 13 34" src="https://user-images.githubusercontent.com/26695477/44244383-b8616f80-a1a1-11e8-887e-82b2f5c8b0be.png">
---------------------------------------------------------------------------------------------------
<img width="363" alt="screenshot 2018-08-16 22 15 02" src="https://user-images.githubusercontent.com/26695477/44244455-14c48f00-a1a2-11e8-8049-b1439f911410.png">

**Comment by Huddie** [comment]
@BrianLitwin I very much agree. I think the Github profile is great for web but cannot be well replicated in mobile. We should look for inspriation outside but try to maintain a familiar interface

**Comment by Huddie** [comment]
Concept
![profile stage 5](https://user-images.githubusercontent.com/10662471/44246492-71c54280-a1ac-11e8-8794-df08a589db19.jpg)

**Comment by Huddie** [comment]
Wasn't going to sketch up 2 new VC's for repos and followers/following so I Just put 2 cells in 1 View.

Cell 1: Repo cell
Cell 2: Following/Followers

![profile stage 4](https://user-images.githubusercontent.com/10662471/44247336-b0a9c700-a1b1-11e8-9032-104673134e2b.jpg)

**Comment by rnystrom** [comment]
Like the directions! I think we should start with what our **information hierarchy** should be before going too deep. There's so many things that could exist on the profile!

- Avatar
- Username
- Display name
- Bio, location, links
- Organizations
- Pinned repos
- Repos
- Followers, following
- Stars
- Activity

Then there are **actions** to take:

- Follow, unfollow
- Share
- Block, report

### Brainstorm

Brainstorming a bit, we can slice this into buckets ranked by priority:

1. **Who?**
  a. Personal identity
    1. Avatar
    2. Username, display name
    3. Bio
    4. Location, employer, links
    5. Stars
  b. Network identity
    1. Followers
    2. Following
    3. Repositories
2. **What do they do?**
  a. Organizations
3. **What have they been doing?**
  a. Pinned Repos
  b. Activity

### Inspiration

Pulling a few images from Dribbble that drew me in:

<img width="300" alt="screen shot 2018-08-17 at 1 50 11 pm" src="https://user-images.githubusercontent.com/739696/44288336-dd4af680-a224-11e8-9408-cf8ebcd78d84.png">

![attachment_2](https://user-images.githubusercontent.com/739696/44288349-e63bc800-a224-11e8-9eae-4ee2c794836d.png)

![user-profile](https://user-images.githubusercontent.com/739696/44288431-2ef38100-a225-11e8-9419-68c272713e64.png)

**Comment by rnystrom** [comment]
Here's a [starter sketch file](https://github.com/GitHawkApp/GitHawk/files/2299561/profile.zip) I whipped up on the flight home yesterday. We should stick to the white background style that we're now using on Issues.

![profile](https://user-images.githubusercontent.com/739696/44299822-c52db280-a2ca-11e8-8336-b58de7159019.png)

**Comment by Huddie** [comment]
Working off @rnystrom sketch file. Merging in some of my original thoughts.

![iphone x](https://user-images.githubusercontent.com/10662471/44304589-fc817b00-a32d-11e8-8176-eb0b5c9a50ac.png)

**Comment by rnystrom** [comment]
This weekend I started putting together a templated Sketch doc to make wireframing easier. Has colors, text, and icons in shareable components: [app.zip](https://github.com/GitHawkApp/GitHawk/files/2302819/app.zip)

It's not totally complete, but flagging as it might help w/ some of the design language.

cc @cieslakdawid 

---

@Huddie some immediate thoughts:

- I'm not a fan of the centered-profile design b/c it wont scale well for iPad w/out massively unused whitespace. I'm all about breathing room, but I think it'll be overboard when scaled up.
- The button designs aren't consistent w/ the rest of the app:
  - Multiple bold/contrasted buttons: why do "Following" and "recent" have the same weight?
  - "Following" should be "Unfollow" (same as GitHub.com)
  - I'll admit the app doesn't have a totally consistent button design either. Should be another template we add in the Sketch file and make everything consistent! Harkens back to the gradient discussion (maybe we go full-gradient buttons like all of GitHub...)
- Do we need form titles for everything? e.g. "bio" probably unecessary
- I want to get rid of having the user handle in the view and just make that the nav item title


**We should boil down what the 1.0 of profile on GitHawk should do.**

What are the most common actions one takes when viewing someone's GitHub profile? For me, I check where they work, find their website/contact info, or see how "notable" they are (followers, big projects they own)

Once we do this, we can focus on only the features that satisfy only those actions and ignore the others for later on. Otherwise we're going to add a lot of complicated features that aren't a priority (e.g. contribution activity).

**Comment by Huddie** [comment]
@rnystrom 

- iPad design
I think on iPhone the centered look is nice. I see what you mean that iPad could seem very open. I've never tried GitHawk on iPad but are layouts always the same? I'd suggest that on iPad the name and handle (or whatever we replace it with) become horizontally rather than vertically stacked 

- Button consistency
I really was trying to give a rough sketch, I'm sure there are plenty of inconsistency ATM but I'll touch it up if we go along this path.

- Form titles
Sections above the "Following" button don't have titles. I don't think it looks great if some sections below have and some don't, trying to keep consistent. I could move the bio above the line but I feel it fits better below.

- Removing handle in view
Sure I can move it out of view. I can replace it with location/email

- Most common actions
I agree there are probably some common ones. I think pinned repos + org. show their notable work. The activity allows you to see what they are currently up to. I think to follow/unfollow is essential along with name + handle and bio and I think quick links to repo/following and followers is appropriate rather than having them actually displayed in the main view.

- Website and Contact
I feel these could be found in an Action Controller or something. I tried to keep the main view populated with things that every GitHub user has. (Organizations and Bio are exceptions). We could always add them in though. 

We could remove contribution activity but it happens to be on the profile page for github mobile and github web. I personally like it but I can see that it isn't necessary V1. 

Let me know your thoughts.

**Comment by Huddie** [comment]
Okay so shift the name and (soon to be added) location + email to horizontal?

**Comment by Huddie** [comment]
Updated. 

<img src="https://user-images.githubusercontent.com/10662471/44360766-e092ec00-a489-11e8-8c03-f5e095e99834.png" width="300" >

**Comment by cieslakdawid** [comment]
I almost missed such a great stuff here <3 

@rnystrom Do you have a plan to work more on the sketch for wireframes before the weekend? 
I underestimated my week and I'm running a bit off the schedule with the `starter pack`, but this Friday I'm back in life(xCode + Sketch ;) ) and want to finish in 2/3 days. To avoid duplication, I'll just post the state of work before starting on Friday.

**Comment by BrianLitwin** [comment]
I think one consideration is that if we're going to have tabs down the road for Repos and Starred Repos, as in Sherlouk's design, the current "Repos/Followers/Following" UI would require a significant re-design as it is now: 

<img width="232" alt="screenshot 2018-08-20 15 20 39" src="https://user-images.githubusercontent.com/26695477/44361763-e2aa7a00-a48c-11e8-9622-af0834962c67.png">

Not trying to look ahead too much, but the more prominent that UI is in our 1.0, the trickier it will be to de-couple later on. 

Would also like to note, unless the Repos and Starred Repos tab are meant to look different from the Search Repos ViewController we currently use, we can reuse SearchRepoResult + SearchRepoResultSectionController, which makes those tabs very easy to build. See [this view controller](https://github.com/GitHawkApp/GitHawk/blob/6e1bd75f17883fbbfbc8013f9364f10d3a695f35/Classes/User%20Profile/Profile%20ViewControllers/UserProfileRepositoriesViewController.swift), which can be used for both Repos and Starred. 

I don't see a great reason to spend time building links to a GitHub web view when we already have so much of the architecture to display them natively already in place. More importantly, if we put Repos front and center, we'll have to re-design down the road if we ever want to give Repos a tab.

**Comment by Huddie** [comment]
I think the Repo/Followers/Following buttons should take you to a new view rather than have them as tabs. And the new view should be native for sure.

**Comment by Sherlouk** [comment]
Just saying that this all looks beautiful and is far better than my envelope drawing 😂

**Comment by Huddie** [comment]
@Sherlouk You got it started. Gotta start somewhere. Thank you

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by rnystrom** [comment]
@cieslakdawid not really, pretty slammed the next two weeks then going out of town.

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by Huddie** [comment]
Any thought on finishing up this design so @BrianLitwin can tweak his profile branch? @rnystrom Any suggestions for improvement on the last design render above?

**Comment by BrianLitwin** [comment]
@Huddie lemme make a Sketch prototype of @Sherlouk 's original design and so we can put that into the discussion. I'll do that in the next few days.

**Comment by Huddie** [comment]
Sure sounds good. 👌🏻 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by BrianLitwin** [comment]
Still not a Sketch pro so the best I can do for a screenshot for now:
 
<img width="292" alt="screenshot 2018-10-02 16 54 40" src="https://user-images.githubusercontent.com/26695477/46376443-02c08200-c664-11e8-8564-db9a69b19c75.png">

**Comment by Huddie** [comment]
@BrianLitwin looks good to me 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by rnystrom** [comment]
@BrianLitwin I really dig it!!

**Comment by Huddie** [comment]
Maybe follow/unfollow is a navbar item or shows up in a action controller? 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by Sherlouk** [comment]
Interesting looking envelope 😂 Looks great!

Think I had planned to have a ... menu in the navbar which does things like follow/unfollow and share

**Comment by ijm8710** [comment]
This is all really cool. Thanks for tagging this ticket @brianlitwin

No rush at all, but goal would be to have pulse activity for both user (including perhaps a feed devoted to just posts and comments) and repository

**Comment by Sherlouk** [comment]
Is anybody working on this at the moment?

Might try and get a bare bones MVP going - think once the foundation is laid we can start to build on top of that with more tabs and information!

**Comment by BrianLitwin** [comment]
@Sherlouk I've taken soft passes in #2125 and #2113 - I think the client/graphql stuff from those pr's is valid. If I post that part (couple hundred lines), can y'all check out the pr and build on top of it?

**Comment by BrianLitwin** [comment]
@Sherlouk okay lemme clean up the UserProfile branch then, I don't think that's up to date

**Comment by kocheck** [comment]
Hey Folks 👋 , I am a designer and wanted to help out.

@Sherlouk and @BrianLitwin are the [Designs posted Oct 2 ](https://github.com/GitHawkApp/GitHawk/issues/2106#issuecomment-426427887) What you are currently working off of? 
Can I provide any design support to help ya'll out? 

#### Right now I think I can work on a rough design for;
- Repos tab
- Stars tab
- Share / Follow Drawer?
- Expanded Pined Repos

---
Let me know how I can help!

**Comment by BrianLitwin** [comment]
@kocheck don't feel constrained by any of the designs on this thread; there's no consensus that I know of. If you have other ideas, feel free to share them.

**Comment by kocheck** [comment]
Roger that! I wanted to try and take a good crack at it this weekend thanks!

**Comment by kocheck** [comment]
Hey, Folks 👋  Found some time to jump into the design file this morning. How do ya'll feel about this direction? Any comments or feedback?

 ## Design Overview 1
The closest to the last posted designs. 

### Updates
- ViewController Tab is moved down to provide a little more hierarchy.
- Included a version with (aprox) max character count for the bio. 
- Moved the Follow and Share actions into an Action Sheet triggered via the Menu icon in the Navbar

| Short Bio | Long Bio | iPhone SE |
| :---: | :---: | :---: |
| <img src="https://user-images.githubusercontent.com/7952000/48314815-0c130780-e59d-11e8-8d80-0704a0bd067a.png" width="300" />  | <img src="https://user-images.githubusercontent.com/7952000/48314816-0c130780-e59d-11e8-926a-6b45d69d357b.png" width="300" /> | <img src="https://user-images.githubusercontent.com/7952000/48314817-0c130780-e59d-11e8-95de-83ea43ce3179.png" width="300" /> |


## Design Overview 2
This design steps out just a little further. If ya'll want me to keep going and head overboard let me know 😄 

## Updates
- ViewController Tab is moved down to provide a little more hierarchy.
- Included (aprox) max character count for the bio. 
    - Moved bio to be inside the table view.
- Did a quick pass at Profile Stats 
    - Repos
    - Followers
    - Following
- Added Follow/unFollow button
- Moved the Follow and Share actions into an Action Sheet triggered via menu icon next to follow button

| Unfollow | Follow | iPhone SE |
| :---: | :---: | :---: |
| <img src="https://user-images.githubusercontent.com/7952000/48314865-affcb300-e59d-11e8-99b6-2bf9ea943862.png" width="300" />  | <img src="https://user-images.githubusercontent.com/7952000/48314866-affcb300-e59d-11e8-881c-41d79d9943b0.png" width="300" /> | <img src="https://user-images.githubusercontent.com/7952000/48314868-affcb300-e59d-11e8-9111-f0b8fcba9f61.png" width="300" /> |



## Still Designing / Next Steps
- I would like to explore a version where the pinned Repos are contained within a box. [Like in this comment ](https://github.com/GitHawkApp/GitHawk/issues/2106#issuecomment-414426871) that @Huddie created. I rather like that. 
- pulse activity for the user `*`
- pulse activity for repository `*`
- Expanded states. for items in the table view
- Repos (tab)
- Stars (tab) 
- ActionSheet

### Unknowns
- I am making an assumption Githawk supports iPhone SE with the auto layout tool, from the conversation above 👆talking about iPad support.
- I am also making a few Design assumptions here and there let me know if you spot anything drastically horrendous. 

- `*` for the pulse activity, I have a stupid question. Would Githawk display these items just like Github or would this be more custom?

**Comment by Huddie** [comment]
Totally agree moving down the tabbar  looks much better +1

For second design: 

- Follow/Unfollow I feel is a 1 maybe 2 time action. It seems to take up a big part of the top view and probably will never change for most people once clicked.
- Repos seem to come up twice (tab and in the top part)
- I personally think the bio is attached to name, location etc and should no be in the lower section when the others are in the top section. 

I like both for the most part. I think design 1 short bio is a good place to start and design 2 can be a more fleshed our second version 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by rnystrom** [comment]
@kocheck yo that looks SO GOOD! Love moving the bar down, very clean!

@Huddie @BrianLitwin playing devil's advocate w/ myself, maybe the following/etc count _shouldn't_ be so prominent (like option 1) since GitHub is about doing work and less about who's-who. But I think it's a useful metric for like "social capital" when meeting new people. 

I also really love the idea of moving the Follow into the action sheet. Remove the social aspects and keep it focused on "who is this person and what work do they do" vs social networking.

I know I'm kind of churning myself here, but once I saw design 1 and compared it w/ 2, I felt like 2 is too much Twitter/Instagram and less _work_.

Thoughts?

**Comment by flovilmart** [comment]
I also lean towards the professional aspect of 1 vs the social emphasis of 2 as @rnystrom.

**Comment by jdisho** [comment]
I am more for the 1st option 😄. Points more “important” information regarding what the person is doing and who is this, rather than social networking (2nd one looks like Insta). 
Additionally, I would suggest to leave the bio out and make more space for the pager info. 
Also, I would have switched places between name/username and location/company/website. For me is more intuitive to see the name/username in the right side of the profile photo. (like the version that @BrianLitwin suggested above) 😊

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by rnystrom** [comment]
@kocheck so cool that you’re spinning through all these edge cases!

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by j-f1** [comment]
Could the entire screen scroll so the tabs become a fixed  header when you scroll far enough? That would mean the entire bio would be visible on all tabs and that the entire screen can be filled with repos if needed.

**Comment by kocheck** [comment]
@j-f1 That's what I am thinking. 👍 

I might swape the Name/Username with the Location, company, and contact info again. Take advantage of the screen height.

**Comment by kocheck** [comment]
Traveling for the holidays, and Had an idea while in the car yesterday. Mocked it up really quick this morning. How do ya'll feel about moving the profile over to the right?

- The Info can flow down, as the user adds or removes more. 
- On iPad, would it cause issues to make the profile image bigger? Help fill that empty space on the right. 



<img width="949" alt="screen shot 2018-11-22 at 8 59 15 am" src="https://user-images.githubusercontent.com/7952000/48907411-544fe680-ee35-11e8-9053-daa8fa7e0a65.png">


---

![screen shot 2018-11-22 at 8 59 31 am](https://user-images.githubusercontent.com/7952000/48907408-53b75000-ee35-11e8-8217-81c96de40071.png)
![screen shot 2018-11-22 at 8 59 26 am](https://user-images.githubusercontent.com/7952000/48907410-53b75000-ee35-11e8-8691-d0f6bbe97540.png)

**Comment by rnystrom** [comment]
@kocheck I think that looks _fantastic_. Love the icon on the right. Makes the whole interface very legible.

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by ijm8710** [comment]
This is awesome! Revisiting kochecks question the other week, I think he asked about pulse activity sharing github or appearing more custom, not sure if that was ever answered which way the team was leaning and curious myself...

Kocheck, would you envision activity being below pinned, as a fourth tab, an overflow option? Unlike followers which you guys realized was ultimately not a pivotal view, to me at least, activity is the most important view on a profile.

**Comment by kocheck** [comment]
@ijm8710 I still need to look through the GitHub docs but if we can display that data I would love to drop it at the top of the Overview Tab. 👍

If we feel good about this as a solution, what would the best way to ship, or submit it to ya'll? A Zeplin project, or should I do a Pull request to the design repo with the updated sketch file? 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by jdisho** [comment]
Slick 🤩!!! @kocheck

**Comment by j-f1** [comment]
@kocheck api.github.com/users/:user/events

**Comment by rnystrom** [comment]
@kocheck PR to design repo would be 👌

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by TigPT** [comment]
None of the images on this thread are up. Anyone have them and can upload them as a GitHub attachment?

**Comment by kocheck** [comment]
Designs were merged into the master sketch file I believe @TigPT 

https://github.com/GitHawkApp/Design-Resources

**Comment by kocheck** [comment]
@TigPT Let me know if you need/want a different file type and I can export that out for you.

**Comment by TigPT** [comment]
> Designs were merged into the master sketch file I believe @TigPT
> 
> https://github.com/GitHawkApp/Design-Resources

And images are now loading, looks like it was a network problem on my side and I assumed wrongly that they are offline.

Thanks for pointing out to Design repo.

**Comment by Huddie** [comment]
New designs coming? Excited!
